cli/daemon/dash/dashapp/src/components/api/SchemaView.tsx

```diff
@@ -131,207 +220,173 @@ class GoDialect extends DialectIface {
-    </>
+  protected renderMap(t: MapType) {
+    this.writeln("{")
+    this.level++
+    this.indent()
+    this.writeType(t.key)
+    this.write(": ")
+    this.writeType(t.value)
+    this.writeln()
+    this.write("}")
   }
 
-  renderList(t: ListType, level: number) {
-    return <>
-      {"["}
-      {this.renderType(t.elem, level)}
-      {"]"}
-    </>
+  protected renderList(t: ListType) {
+    this.write("[")
+    this.writeType(t.elem)
+    this.write("]")
   }
 
-  renderBuiltin(t: BuiltinType, level: number) {
+  protected renderBuiltin(t: Builtin) {
     switch (t) {
-    case BuiltinType.Any: return "<any>"
-    case BuiltinType.Bool: return "true"
-    case BuiltinType.Int: return "1"
-    case BuiltinType.Int8: return "1"
-    case BuiltinType.Int16: return "1"
-    case BuiltinType.Int32: return "1"
-    case BuiltinType.Int64: return "1"
-    case BuiltinType.Uint: return "1"
-    case BuiltinType.Uint8: return "1"
-    case BuiltinType.Uint16: return "1"
-    case BuiltinType.Uint32: return "1"
-    case BuiltinType.Uint64: return "1"
-    case BuiltinType.Float32: return "2.3"
-    case BuiltinType.Float64: return "2.3"
-    case BuiltinType.String: return "\"some-string\""
-    case BuiltinType.Bytes: return "\"base64-encoded-bytes\""
-    case BuiltinType.Time: return "\"2009-11-10T23:00:00Z\""
-    case BuiltinType.UUID: return "\"7d42f515-3517-4e76-be13-30880443546f\""
-    case BuiltinType.JSON: return "{\"some-json-data\": true}"
-    case BuiltinType.USER_ID: return "\"some-user-id\""
-    default: return "<unknown>"
+      case Builtin.ANY: return this.write("<any data>")
+      case Builtin.BOOL: return this.write("true")
+      case Builtin.INT: return this.write("1")
+      case Builtin.INT8: return this.write("1")
+      case Builtin.INT16: return this.write("1")
+      case Builtin.INT32: return this.write("1")
+      case Builtin.INT64: return this.write("1")
+      case Builtin.UINT: return this.write("1")
+      case Builtin.UINT8: return this.write("1")
+      case Builtin.UINT16: return this.write("1")
+      case Builtin.UINT32: return this.write("1")
+      case Builtin.UINT64: return this.write("1")
+      case Builtin.FLOAT32: return this.write("2.3")
+      case Builtin.FLOAT64: return this.write("2.3")
+      case Builtin.STRING: return this.write("\"some string\"")
+      case Builtin.BYTES: return this.write("\"YmFzZTY0Cg==\"") // base64
+      case Builtin.TIME: return this.write("\"2009-11-10T23:00:00Z\"")
+      case Builtin.UUID: return this.write("\"7d42f515-3517-4e76-be13-30880443546f\"")
+      case Builtin.JSON: return this.write("{\"some json data\": true}")
+      case Builtin.USER_ID: return this.write("userID")
+      case Builtin.UNRECOGNIZED: return this.write("<unknown>")
     }
+
+    return unreachableUnknownType(t)
   }
 }
 
 class TableDialect extends DialectIface {
-  render(d: Decl) {
-    const st = d.type.struct
+  typeArgumentStack: Type[][] = [];
+
+  render(d: Type) {
```

Comment:
Yeah this is due to the use of this type, but again I think we should be ok because the metadata schema wouldn't generate on the backend if it wasn't a named type with a struct.